app/src/main/java/ch/epfl/qedit/view/Online/OnlineFragment.java

```diff
@@ -0,0 +1,201 @@
+                    }
+
+                    @Override
+                    public void onScrollStateChanged(
+                            @NonNull RecyclerView recyclerView, int newState) {
+                        super.onScrollStateChanged(recyclerView, newState);
+                        if (newState == RecyclerView.SCROLL_STATE_IDLE) {}
+                    }
+                });
+        // The progress bar is needed while waiting from the database
+        progressBar = view.findViewById(R.id.quiz_loading);
+
+        // Instantiate Handler and the DatabaseService
+        db = DatabaseFactory.getInstance();
+        handler = new Handler();
+        return view;
+    }
+
+    // This function is used to create the list of quizzes for the given user
+    private void createAdapter() {
+        // Retrieve the quizzes from the user
+        quizzes.e = new ArrayList<>();
+
+        // Create the list adapter
+        listAdapter = new ListEditView.Adapter<>(quizzes, item -> item.getValue());
+
+        // Listen to the data changes
+        listAdapter.setItemListener(
+                (position, type) -> {
+                    switch (type) {
+                        case EditRequest:
+                            editQuiz(position);
+                            break;
+                        default:
+                            break;
+                    }
+                });
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.search, menu);
+        MenuItem item = menu.findItem(R.id.app_bar_search);
+        SearchView searchView = (SearchView) item.getActionView();
+        searchView.setOnQueryTextListener(
+                new SearchView.OnQueryTextListener() {
+                    @Override
+                    public boolean onQueryTextSubmit(String query) {
+                        try {
+                            listAdapter.clear();
+                            for (Map.Entry<String, String> e :
+                                    db.searchDatabase(load, load + 10, query).get())
+                                listAdapter.addItem(e);
+                            load = 10;
+                            searchView.clearFocus();
+                        } catch (Exception e) {
+                        }
+
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onQueryTextChange(String newText) {
+                        load = 0;
+                        return false;
+                    }
+                });
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    // This method will be called when an item of the top bar is clicked on
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                requireActivity().onBackPressed();
```

Comment:
There is a bug: when navigating back to the home activity, the hamburger menu item isn't selected. I think that we need to do something like `navigationView.setCheckedItem(id);`.